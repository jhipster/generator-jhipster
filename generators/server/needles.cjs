/**
 * Copyright 2013-2022 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project, see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const { createBaseNeedle } = require('../../lib/support/needles.cjs');

/**
 * @typedef {Object} ApplicationPropertiesNeedles - creates a new type named 'SpecialType'
 * @property {string} [property] - property declaration
 * @property {string} [propertyGetter] - property getter
 * @property {string} [propertyClass] - property class
 */

/**
 * Insert content into ApplicationProperties class
 * @example
 * insertContentIntoApplicationProperties(generator, {
 *   property: 'private final bar = new Bar();',
 *   proppertyGetter: `
 * public getBar() {
 *     return bar;
 * }`,
 *   propertyClass: `
 * public static class Async {
 *     private String foo = "default";
 * }`,
 * })
 * @example
 * generator.editFile(
 *   'ApplicationProperties.java',
 *   insertContentIntoApplicationProperties({
 *     property: 'private final bar = new Bar();',
 *     proppertyGetter: `
 * public getBar() {
 *     return bar;
 * }`,
 *     propertyClass: `
 * public static class Async {
 *     private String foo = "default";
 * }`,
 *   });
 * );
 * @param {import('../generator-base.js')} [generator]
 * @param {ApplicationPropertiesNeedles} needles
 * @returns {string | import('../../generators/generator-base.js').EditFileCallback}
 */
const insertContentIntoApplicationProperties = (generator, needles) => {
  if (!needles) {
    needles = generator;
    generator = null;
  }

  return createBaseNeedle(
    {
      generator,
      filePath: generator ? `${generator.SERVER_MAIN_SRC_DIR}${generator.javaDir}config/ApplicationProperties.java` : undefined,
      needlesPrefix: 'application-properties',
    },
    needles
  );
};

module.exports = {
  insertContentIntoApplicationProperties,
};
