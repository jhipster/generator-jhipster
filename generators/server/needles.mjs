/**
 * Copyright 2013-2023 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project, see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { createBaseNeedle } from '../../lib/support/needles.mjs';

/**
 * @typedef {Object} ApplicationPropertiesNeedles - creates a new type named 'SpecialType'
 * @property {string} [property] - property declaration
 * @property {string} [propertyGetter] - property getter
 * @property {string} [propertyClass] - property class
 */

/**
 * Insert content into ApplicationProperties class
 * @example
 * insertContentIntoApplicationProperties(generator, application, {
 *   property: 'private final bar = new Bar();',
 *   proppertyGetter: `
 * public getBar() {
 *     return bar;
 * }`,
 *   propertyClass: `
 * public static class Async {
 *     private String foo = "default";
 * }`,
 * })
 * @example
 * generator.editFile(
 *   'ApplicationProperties.java',
 *   insertContentIntoApplicationProperties({
 *     property: 'private final bar = new Bar();',
 *     proppertyGetter: `
 * public getBar() {
 *     return bar;
 * }`,
 *     propertyClass: `
 * public static class Async {
 *     private String foo = "default";
 * }`,
 *   });
 * );
 * @param {import('../generator-base.js')} [generator]
 * @param {any} context
 * @param {ApplicationPropertiesNeedles} needles
 * @returns {import('../generator-base.js').CascatedEditFileCallback | import('../generator-base.js').EditFileCallback}
 */
// eslint-disable-next-line import/prefer-default-export
export const insertContentIntoApplicationProperties = (generator, data, needles) => {
  if (!needles) {
    needles = generator;
    generator = null;
  }

  return createBaseNeedle(
    {
      generator,
      filePath: generator ? `${data.SERVER_MAIN_SRC_DIR}${data.javaDir}config/ApplicationProperties.java` : undefined,
      needlesPrefix: 'application-properties',
    },
    needles
  );
};
