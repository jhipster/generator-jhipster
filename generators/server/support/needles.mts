/**
 * Copyright 2013-2023 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project, see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import CoreGenerator from '../../base-core/index.mjs';
import { createBaseNeedle } from '../../base/support/needles.mjs';
import { SpringBootApplication } from '../types.mjs';

type ApplicationPropertiesNeedles = {
  property?: string;
  propertyGetter?: string;
  propertyClass?: string;
};

/**
 * Insert content into ApplicationProperties class
 * @example
 * insertContentIntoApplicationProperties.call(generator, application, {
 *   property: 'private final bar = new Bar();',
 *   proppertyGetter: `
 * public getBar() {
 *     return bar;
 * }`,
 *   propertyClass: `
 * public static class Async {
 *     private String foo = "default";
 * }`,
 * })
 * @example
 * generator.editFile(
 *   'ApplicationProperties.java',
 *   insertContentIntoApplicationProperties({
 *     property: 'private final bar = new Bar();',
 *     proppertyGetter: `
 * public getBar() {
 *     return bar;
 * }`,
 *     propertyClass: `
 * public static class Async {
 *     private String foo = "default";
 * }`,
 *   });
 * );
 */
// eslint-disable-next-line import/prefer-default-export
export function insertContentIntoApplicationProperties(
  this: CoreGenerator | void,
  application: SpringBootApplication,
  needles: ApplicationPropertiesNeedles
) {
  if (this) {
    return createBaseNeedle.call(
      this,
      {
        filePath: `${application.javaPackageSrcDir}config/ApplicationProperties.java`,
        needlesPrefix: 'application-properties',
      },
      needles
    );
  }
  return createBaseNeedle(
    {
      needlesPrefix: 'application-properties',
    },
    needles
  );
}
