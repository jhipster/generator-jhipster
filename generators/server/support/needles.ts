/**
 * Copyright 2013-2025 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project, see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import assert from 'assert';
import type CoreGenerator from '../../base-core/index.js';
import { createBaseNeedle } from '../../base/support/needles.js';

export type ApplicationPropertiesNeedles = {
  property?: string;
  propertyGetter?: string;
  propertyClass?: string;
};

/**
 * Insert content into ApplicationProperties class
 * @example
 * insertContentIntoApplicationProperties.call(generator, application, {
 *   property: 'private final bar = new Bar();',
 *   proppertyGetter: `
 * public getBar() {
 *     return bar;
 * }`,
 *   propertyClass: `
 * public static class Async {
 *     private String foo = "default";
 * }`,
 * })
 * @example
 * generator.editFile(
 *   'ApplicationProperties.java',
 *   insertContentIntoApplicationProperties({
 *     property: 'private final bar = new Bar();',
 *     propertyGetter: `
 * public getBar() {
 *     return bar;
 * }`,
 *     propertyClass: `
 * public static class Async {
 *     private String foo = "default";
 * }`,
 *   });
 * );
 */
export function insertContentIntoApplicationProperties(needles: ApplicationPropertiesNeedles);
export function insertContentIntoApplicationProperties(
  this: CoreGenerator,
  application: { javaPackageSrcDir: string },
  needles: ApplicationPropertiesNeedles,
);
export function insertContentIntoApplicationProperties(
  this: CoreGenerator | void,
  application: { javaPackageSrcDir: string } | ApplicationPropertiesNeedles,
  needles?: ApplicationPropertiesNeedles,
) {
  if (needles) {
    assert.ok(this, 'Generator context is required');

    return createBaseNeedle.call(
      this,
      {
        filePath: `${(application as { javaPackageSrcDir: string }).javaPackageSrcDir}config/ApplicationProperties.java`,
        needlesPrefix: 'application-properties',
      },
      needles,
    );
  }
  return createBaseNeedle(
    {
      needlesPrefix: 'application-properties',
    },
    application as ApplicationPropertiesNeedles,
  );
}
