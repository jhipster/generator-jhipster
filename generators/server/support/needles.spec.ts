/**
 * Copyright 2013-2024 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project, see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { before, it, describe, expect } from 'esmocha';
import { defaultHelpers as helpers, runResult } from '../../../testing/index.js';
import { GENERATOR_SPRING_BOOT } from '../../generator-list.js';
import { insertContentIntoApplicationProperties } from './needles.js';
import type { SpringBootApplication } from '../types.js';

describe('generator - server - support - needles', () => {
  describe('generated project', () => {
    before(async () => {
      await helpers
        .runJHipster(GENERATOR_SPRING_BOOT)
        .withMockedGenerators(['jhipster:common', 'jhipster:languages', 'jhipster:liquibase']);
    });

    it('should match state snapshot', () => {
      expect(runResult.getStateSnapshot()).toMatchSnapshot();
    });

    describe('insertContentIntoApplicationProperties needle', () => {
      it('with a non existing needle', () => {
        const application: SpringBootApplication = runResult.generator.sharedData.getApplication();
        expect(() => insertContentIntoApplicationProperties.call(runResult.generator, application, { foo: 'foo' })).toThrow(
          /Missing required jhipster-needle application-properties-foo not found at/,
        );
      });

      it('without a needle', () => {
        const application = runResult.generator.sharedData.getApplication();
        expect(() => insertContentIntoApplicationProperties.call(runResult.generator, application, {})).toThrow(
          /At least 1 needle is required/,
        );
      });

      describe('when applied', () => {
        const fileRegexp = /config\/ApplicationProperties.java/;
        const property = 'private Foo foo;';
        const propertyGetter = `

    private Foo getFoo() {
        return foo;
    };`;
        const propertyClass = `

        public static Foo{} {
            private String bar;

            public String getBar() {
              return bar;
            }
        };`;
        let snapshot;

        before(() => {
          const application = runResult.generator.sharedData.getApplication();
          insertContentIntoApplicationProperties.call(runResult.generator, application, {
            property,
            propertyGetter,
            propertyClass,
          });
          snapshot = runResult.getSnapshot(file => fileRegexp.test(file.path));
        });

        it('should match snapshot', () => {
          expect(snapshot).toMatchInlineSnapshot(`
{
  "src/main/java/com/mycompany/myapp/config/ApplicationProperties.java": {
    "contents": "package com.mycompany.myapp.config;

import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * Properties specific to JHipster.
 * <p>
 * Properties are configured in the {@code application.yml} file.
 * See {@link tech.jhipster.config.JHipsterProperties} for a good example.
 */
@ConfigurationProperties(prefix = "application", ignoreUnknownFields = false)
public class ApplicationProperties {

    private Foo foo;
    // jhipster-needle-application-properties-property


    private Foo getFoo() {
        return foo;
    };
    // jhipster-needle-application-properties-property-getter


    public static Foo{} {
        private String bar;

        public String getBar() {
          return bar;
        }
    };
    // jhipster-needle-application-properties-property-class
}
",
    "state": "modified",
    "stateCleared": "modified",
  },
}
`);
        });

        it('should not be add the content at second call', () => {
          const application = runResult.generator.sharedData.getApplication();
          insertContentIntoApplicationProperties.call(runResult.generator, application, {
            property,
            propertyGetter,
            propertyClass,
          });
          expect(runResult.getSnapshot(file => fileRegexp.test(file.path))).toEqual(snapshot);
        });

        it('should not be add new content with prettier differences', () => {
          const application = runResult.generator.sharedData.getApplication();
          insertContentIntoApplicationProperties.call(runResult.generator, application, {
            property: '  private   Foo   foo;',
          });
          expect(runResult.getSnapshot(file => fileRegexp.test(file.path))).toEqual(snapshot);
        });

        it('should not be add new content with prettier differences and new lines', () => {
          const application = runResult.generator.sharedData.getApplication();
          insertContentIntoApplicationProperties.call(runResult.generator, application, {
            property: `  private Foo getFoo() {

        return foo;

    };
`,
          });
          expect(runResult.getSnapshot(file => fileRegexp.test(file.path))).toEqual(snapshot);
        });
      });
    });
  });
});
