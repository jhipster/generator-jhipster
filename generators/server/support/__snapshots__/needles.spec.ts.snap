// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator - server - support - needles generated project should match state snapshot 1`] = `
{
  ".devcontainer/Dockerfile": {
    "stateCleared": "modified",
  },
  ".devcontainer/devcontainer.json": {
    "stateCleared": "modified",
  },
  ".mvn/jvm.config": {
    "stateCleared": "modified",
  },
  ".mvn/wrapper/maven-wrapper.jar": {
    "stateCleared": "modified",
  },
  ".mvn/wrapper/maven-wrapper.properties": {
    "stateCleared": "modified",
  },
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "checkstyle.xml": {
    "stateCleared": "modified",
  },
  "mvnw": {
    "stateCleared": "modified",
  },
  "mvnw.cmd": {
    "stateCleared": "modified",
  },
  "npmw": {
    "stateCleared": "modified",
  },
  "npmw.cmd": {
    "stateCleared": "modified",
  },
  "package.json": {
    "stateCleared": "modified",
  },
  "pom.xml": {
    "stateCleared": "modified",
  },
  "src/main/docker/app.yml": {
    "stateCleared": "modified",
  },
  "src/main/docker/grafana/provisioning/dashboards/JVM.json": {
    "stateCleared": "modified",
  },
  "src/main/docker/grafana/provisioning/dashboards/dashboard.yml": {
    "stateCleared": "modified",
  },
  "src/main/docker/grafana/provisioning/datasources/datasource.yml": {
    "stateCleared": "modified",
  },
  "src/main/docker/jhipster-control-center.yml": {
    "stateCleared": "modified",
  },
  "src/main/docker/jib/entrypoint.sh": {
    "stateCleared": "modified",
  },
  "src/main/docker/monitoring.yml": {
    "stateCleared": "modified",
  },
  "src/main/docker/postgresql.yml": {
    "stateCleared": "modified",
  },
  "src/main/docker/prometheus/prometheus.yml": {
    "stateCleared": "modified",
  },
  "src/main/docker/services.yml": {
    "stateCleared": "modified",
  },
  "src/main/docker/sonar.yml": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/ApplicationWebXml.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/GeneratedByJHipster.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/JhipsterApp.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/aop/logging/LoggingAspect.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/aop/logging/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/ApplicationProperties.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/AsyncConfiguration.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/CRLFLogConverter.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/CacheConfiguration.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/Constants.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/DatabaseConfiguration.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/DateTimeFormatConfiguration.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/JacksonConfiguration.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/LoggingAspectConfiguration.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/LoggingConfiguration.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/SecurityConfiguration.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/SecurityJwtConfiguration.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/StaticResourcesWebConfiguration.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/WebConfigurer.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/domain/AbstractAuditingEntity.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/domain/Authority.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/domain/User.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/domain/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/management/SecurityMetersService.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/management/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/repository/AuthorityRepository.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/repository/UserRepository.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/repository/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/security/AuthoritiesConstants.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/security/DomainUserDetailsService.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/security/SecurityUtils.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/security/SpringSecurityAuditorAware.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/security/UserNotActivatedException.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/security/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/service/EmailAlreadyUsedException.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/service/InvalidPasswordException.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/service/MailService.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/service/UserService.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/service/UsernameAlreadyUsedException.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/service/dto/AdminUserDTO.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/service/dto/PasswordChangeDTO.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/service/dto/UserDTO.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/service/dto/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/service/mapper/UserMapper.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/service/mapper/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/service/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/filter/SpaWebFilter.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/filter/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/AccountResource.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/AuthenticateController.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/AuthorityResource.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/PublicUserResource.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/UserResource.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/errors/BadRequestAlertException.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/errors/EmailAlreadyUsedException.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/errors/ErrorConstants.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/errors/ExceptionTranslator.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/errors/FieldErrorVM.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/errors/InvalidPasswordException.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/errors/LoginAlreadyUsedException.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/errors/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/vm/KeyAndPasswordVM.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/vm/LoginVM.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/vm/ManagedUserVM.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/vm/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/resources/banner.txt": {
    "stateCleared": "modified",
  },
  "src/main/resources/config/application-dev.yml": {
    "stateCleared": "modified",
  },
  "src/main/resources/config/application-prod.yml": {
    "stateCleared": "modified",
  },
  "src/main/resources/config/application-tls.yml": {
    "stateCleared": "modified",
  },
  "src/main/resources/config/application.yml": {
    "stateCleared": "modified",
  },
  "src/main/resources/i18n/messages.properties": {
    "stateCleared": "modified",
  },
  "src/main/resources/logback-spring.xml": {
    "stateCleared": "modified",
  },
  "src/main/resources/templates/error.html": {
    "stateCleared": "modified",
  },
  "src/main/resources/templates/mail/activationEmail.html": {
    "stateCleared": "modified",
  },
  "src/main/resources/templates/mail/creationEmail.html": {
    "stateCleared": "modified",
  },
  "src/main/resources/templates/mail/passwordResetEmail.html": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/IntegrationTest.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/TechnicalStructureTest.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/config/AsyncSyncConfiguration.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/config/CRLFLogConverterTest.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/config/EmbeddedSQL.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/config/PostgreSqlTestContainer.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/config/SpringBootTestClassOrderer.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/config/SqlTestContainer.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/config/SqlTestContainersSpringContextCustomizerFactory.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/config/StaticResourcesWebConfigurerTest.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/config/WebConfigurerTest.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/config/WebConfigurerTestController.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/config/timezone/HibernateTimeZoneIT.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/domain/AssertUtils.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/domain/AuthorityAsserts.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/domain/AuthorityTest.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/domain/AuthorityTestSamples.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/management/SecurityMetersServiceTests.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/repository/timezone/DateTimeWrapper.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/repository/timezone/DateTimeWrapperRepository.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/security/DomainUserDetailsServiceIT.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/security/SecurityUtilsUnitTest.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/security/jwt/AuthenticationIntegrationTest.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/security/jwt/JwtAuthenticationTestUtils.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/security/jwt/TokenAuthenticationIT.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/security/jwt/TokenAuthenticationSecurityMetersIT.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/service/MailServiceIT.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/service/UserServiceIT.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/service/mapper/UserMapperTest.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/web/filter/SpaWebFilterIT.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/web/rest/AccountResourceIT.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/web/rest/AuthenticateControllerIT.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/web/rest/AuthorityResourceIT.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/web/rest/PublicUserResourceIT.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/web/rest/TestUtil.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/web/rest/UserResourceIT.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/web/rest/WithUnauthenticatedMockUser.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/web/rest/errors/ExceptionTranslatorIT.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/web/rest/errors/ExceptionTranslatorTestController.java": {
    "stateCleared": "modified",
  },
  "src/test/resources/META-INF/spring.factories": {
    "stateCleared": "modified",
  },
  "src/test/resources/config/application-testdev.yml": {
    "stateCleared": "modified",
  },
  "src/test/resources/config/application-testprod.yml": {
    "stateCleared": "modified",
  },
  "src/test/resources/config/application.yml": {
    "stateCleared": "modified",
  },
  "src/test/resources/junit-platform.properties": {
    "stateCleared": "modified",
  },
  "src/test/resources/logback.xml": {
    "stateCleared": "modified",
  },
  "src/test/resources/templates/mail/activationEmail.html": {
    "stateCleared": "modified",
  },
  "src/test/resources/templates/mail/creationEmail.html": {
    "stateCleared": "modified",
  },
  "src/test/resources/templates/mail/passwordResetEmail.html": {
    "stateCleared": "modified",
  },
  "src/test/resources/templates/mail/testEmail.html": {
    "stateCleared": "modified",
  },
}
`;
